pkg/registry/file/vulnsummarystorage_test.go

```diff
@@ -0,0 +1,461 @@
+package file
+
+import (
+	"context"
+	"testing"
+
+	"github.com/kubescape/storage/pkg/apis/softwarecomposition"
+	"github.com/kubescape/storage/pkg/apis/softwarecomposition/v1beta1"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/storage"
+)
+
+func TestVulnSummaryStorageImpl_Count(t *testing.T) {
+	files := []string{
+		"/other/type/ns/titi.json",
+		"/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3filtereds/kubescape/titi.json",
+		"/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3filtereds/other/toto.json",
+		"/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/kubescape/toto.json",
+		"/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/other/toto.json",
+	}
+	tests := []struct {
+		name    string
+		key     string
+		want    int64
+		wantErr bool
+	}{
+		{
+			name: "one object",
+			key:  "/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/kubescape/toto",
+			want: 1,
+		},
+		{
+			name: "one ns",
+			key:  "/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/kubescape",
+			want: 1,
+		},
+		{
+			name: "one type",
+			key:  "/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s",
+			want: 2,
+		},
+		{
+			name: "all types",
+			key:  "/spdx.softwarecomposition.kubescape.io",
+			want: 4,
+		},
+		{
+			name: "from top",
+			key:  "/",
+			want: 5,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fs := afero.NewMemMapFs()
+			_ = fs.Mkdir(DefaultStorageRoot, 0755)
+			for _, f := range files {
+				_ = afero.WriteFile(fs, DefaultStorageRoot+f, []byte(""), 0644)
+			}
+			s := NewStorageImpl(fs, DefaultStorageRoot)
+			got, err := s.Count(tt.key)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Count() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("Count() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestVulnSummaryStorageImpl_Create(t *testing.T) {
+	type args struct {
+		key string
+		obj runtime.Object
+		out runtime.Object
+		in4 uint64
+	}
+	tests := []struct {
+		name     string
+		readonly bool
+		args     args
+		wantErr  bool
+		want     runtime.Object
+	}{
+		{
+			name: "not supported",
+			args: args{
+				key: "/spdx.softwarecomposition.kubescape.io/vulnerabilitysummaries/kubescape/toto",
+				obj: &v1beta1.VulnerabilitySummary{},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fs := afero.NewReadOnlyFs(afero.NewMemMapFs())
+			s := NewVulnSummaryStorageImpl(fs, DefaultStorageRoot)
+			err := s.Create(context.TODO(), tt.args.key, tt.args.obj, tt.args.out, tt.args.in4)
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
```

Comment:
suggestion: we should test for the error that we want instead of some blanket error

How do you know that you got an error for the correct reason? How do you know the error you get is the one that you expect, and not some other error?